documentation/typescript/adding-custom-login-form-with-spring-security.asciidoc

```diff
@@ -0,0 +1,176 @@
+---
+title: Adding a Custom Login Form with Spring Security
+order: 91
+layout: page
+---
+
+ifdef::env-github[:outfilesuffix: .asciidoc]
+
+= Adding a Custom Login Form with Spring Security
+
+Instead of using the <<adding-login-form-with-spring-security#, default Spring login form>>, you may want to use your own customized form. This article describes how to do that.
+
+== Dependencies and Server Configuration
+
+The dependencies and server configuration is the same as described in <<adding-login-form-with-spring-security#, Adding a Login Form with Spring Security>>, with a difference that you need to specify the URL of the custom login view.
+
+.SecurityConfig.java
+[source,java]
+----
+@EnableWebSecurity
+@Configuration
+public class SecurityConfig extends WebSecurityConfigurerAdapter {
+
+  @Override
+  protected void configure(HttpSecurity http) throws Exception {
+    // Vaadin already handles csrf.
+    http.csrf().disable();
+    // specify the URL of the login view, the value of the parameter
+    // is the defined route for the login view component.
+    http.formLogin().loginPage("login");
+  }  
+}
+----
+
+== Custom login view
+
+The easiest way to make a login view is to use the `<vaadin-login-overlay>` component.
+Vaadin provides a `login` helper method for Spring Security based authentication that you can use for the login action.
+
+.frontend/login-view.ts
+[source, typescript]
+----
+@customElement('login-view')
+export class LoginView extends LitElement implements AfterEnterObserver {
+
+  @property({type: Boolean})
+  private error = false;
+
+  @property({type: Boolean})
+  private open = true;
+
+  // the url to redirect to after a successful login
+  private returnUrl = '/';
+
+  private onSuccess: (result:LoginResult) => void;
+
+  constructor(){
+    super();
+    this.onSuccess = () => {
+      Router.go(this.returnUrl);
+    };
+  }
+
+  render() {
+    return html`
+      <vaadin-login-overlay
+        ?opened="${this.open}" 
+        .error=${this.error}
+        @login="${this.login}">    
+      </vaadin-login-overlay>
+    `;
+  }
+
+  async login(event: CustomEvent): Promise<LoginResult> {
+    this.error = false;
+    // use the Vaadin provided login helper method to obtain the LoginResult
+    const result = await login(event.detail.username, event.detail.password);
```

Comment:
Would be nice to have the import line for `login` in this example snippet, so that it is known where it is coming from.

Reply:
good point, added the import line